Fix M_Enemy

AJenbo opened the pull request:
This fixes an issue where the gargoyle would not sleep

galaxyhaxz commented:
Seeing as it uses BYTE1, it should probably be 0x4 << 8 (0x400) intstead:
```
v1->_mFlags |= MFLAG_SEARCH2; // 0x400
```

AJenbo commented:
@galaxyhaxz  Thanks, I knew something was up, but I was to tired to figure it out so just went with a revert :)

I have corrected the flag and updated it's name now that it's more clear what it does.

mewmew commented:
Haha, great that this also fixed a `BUGFIX` comment!

```c
// BUGFIX MFLAG_SEARCH2 appears to never be set, maybe use MFLAG_TARGETS_MONSTER?
```
